pkg/platform/runtime/setup/setup.go

```diff
@@ -160,6 +163,98 @@ func NewWithModel(target Targeter, msgHandler Events, model ModelProvider, an an
 
 // Update installs the runtime locally (or updates it if it's already partially installed)
 func (s *Setup) Update() error {
+	var artifacts []artifact.ArtifactID
+	mutex := sync.Mutex{}
+
+	// Fetch and install each runtime artifact.
+	err := s.fetchArtifacts(func(a artifact.ArtifactID, archivePath string, as ArtifactSetuper) error {
```

Comment:
Can we move the callback into its own function? In terms of reading the diff it'd help if this was close to the original location. I can't easily tell how much of the underlying code here was changed, if at all.

Reply:
I'd prefer not because it's a closure (uses `mutex` and `artifacts`). I could make them belong to a `Setup` object, but that doesn't feel right. I'm open to suggestions.